pysb/tests/test_simulator_bng.py

```diff
@@ -0,0 +1,115 @@
+from pysb.testing import *
+import numpy as np
+from pysb import Monomer, Parameter, Initial, Observable, Rule
+from pysb.simulator.bng import BngSimulator, PopulationMap
+from pysb.bng import generate_equations
+from pysb.examples import robertson, expression_observables, earm_1_0
+
+_BNG_SEED = 123
+
+
+class TestBngSimulator(object):
+    @with_model
+    def setUp(self):
+        Monomer('A', ['a'])
+        Monomer('B', ['b'])
+
+        Parameter('ksynthA', 100)
+        Parameter('ksynthB', 100)
+        Parameter('kbindAB', 100)
+
+        Parameter('A_init', 0)
+        Parameter('B_init', 0)
+
+        Initial(A(a=None), A_init)
+        Initial(B(b=None), B_init)
+
+        Observable("A_free", A(a=None))
+        Observable("B_free", B(b=None))
+        Observable("AB_complex", A(a=1) % B(b=1))
+
+        Rule('A_synth', None >> A(a=None), ksynthA)
+        Rule('B_synth', None >> B(b=None), ksynthB)
+        Rule('AB_bind', A(a=None) + B(b=None) >> A(a=1) % B(b=1), kbindAB)
+
+        self.model = model
+        generate_equations(self.model)
+
+        # Convenience shortcut for accessing model monomer objects
+        self.mon = lambda m: self.model.monomers[m]
+
+        # This timespan is chosen to be enough to trigger a Jacobian evaluation
+        # on the various solvers.
+        self.time = np.linspace(0, 1)
+        self.sim = BngSimulator(self.model, tspan=self.time)
+
+    def test_1_simulation(self):
+        x = self.sim.run()
+        assert np.allclose(x.tout, self.time)
+
+    def test_multi_simulations(self):
+        x = self.sim.run(n_runs=10)
+        assert np.shape(x.observables) == (10, 50)
+        # Check initials are getting correctly reset on each simulation
+        assert np.allclose(x.species[-1][0, :], x.species[0][0, :])
+
+    def test_change_parameters(self):
+        x = self.sim.run(n_runs=10, param_values={'ksynthA': 200},
+                         initials={self.model.species[0]: 100})
+        species = np.array(x.all)
+        assert species[0][0][0] == 100.
+
+    def test_bng_pla(self):
+        self.sim.run(n_runs=5, method='pla', seed=_BNG_SEED)
+
+    def tearDown(self):
+        self.model = None
+        self.time = None
+        self.sim = None
+
+
+def test_bng_ode_with_expressions():
+    model = expression_observables.model
+    model.reset_equations()
+
+    sim = BngSimulator(model, tspan=np.linspace(0, 1))
+    x = sim.run(n_runs=1, method='ode')
+    assert len(x.expressions) == 50
+    assert len(x.observables) == 50
+
+
+def test_nfsim():
+    # Make sure no network generation has taken place
```

Comment:
Does this actually verify that network generation hasn't occurred? I can't see it.